node/impl/full/state.go

```diff
@@ -617,14 +618,22 @@ func (m *StateModule) StateWaitMsg(ctx context.Context, msg cid.Cid, confidence
 
 		t, err := stmgr.GetReturnType(ctx, m.StateManager, vmsg.To, vmsg.Method, ts)
 		if err != nil {
-			return nil, xerrors.Errorf("failed to get return type: %w", err)
-		}
-
-		if err := t.UnmarshalCBOR(bytes.NewReader(recpt.Return)); err != nil {
-			return nil, err
+			if errors.Is(err, stmgr.ErrMetadataNotFound) {
+				// This is not nececessary an error -- EVM methods (and in the future native actors) may
```

Comment:
```suggestion
				// This is not necessarily an error -- EVM methods (and in the future native actors) may
```

Reply:
Fixed in https://github.com/filecoin-project/lotus/pull/9984.